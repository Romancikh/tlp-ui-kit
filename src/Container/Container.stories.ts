import type { Meta, StoryObj } from "@storybook/react";

import Container from "./Container";

const meta = {
  component: Container,
  tags: ["autodocs"],
  title: "Container",
} satisfies Meta<typeof Container>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    background: "lightgreen",
    children: "Some content",
    height: "100vh",
  },
};

export const MaxWidth: Story = {
  args: {
    background: "lightblue",
    children: "Some content",
    height: "100vh",
    maxWidth: false,
  },
};
