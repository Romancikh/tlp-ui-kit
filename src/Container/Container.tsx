import MuiContainer from "@mui/material/Container";
import { PropsWithChildren } from "react";

type ContainerProps = PropsWithChildren & {
  alignItems?: "center";
  background?: string;
  display?: "flex" | "grid";
  flexDirection?: "column";
  height?: string;
  margin?: string;
  maxWidth?: "xl" | false;
  minHeight?: string;
  placeItems?: "center";
};

function Container({
  alignItems,
  background,
  children,
  display,
  flexDirection,
  height,
  margin,
  maxWidth,
  minHeight,
  placeItems,
}: ContainerProps) {
  return (
    <MuiContainer
      maxWidth={maxWidth}
      sx={{
        alignItems,
        background,
        display,
        flexDirection,
        height,
        margin,
        minHeight,
        placeItems,
      }}
    >
      {children}
    </MuiContainer>
  );
}

export default Container;
