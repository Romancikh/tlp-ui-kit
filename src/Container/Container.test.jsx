import { render } from "@testing-library/react";
import Container from "./Container";

test("Default container renders correctly", () => {
  const { container } = render(
    <Container background="lightgreen" height="100vh" children="Some content" />
  );
  expect(container).toMatchSnapshot();
});

test("MaxWidth container renders correctly", () => {
  const { container } = render(
    <Container
      maxWidth={false}
      background="lightblue"
      height="100vh"
      children="Some content"
    />
  );
  expect(container).toMatchSnapshot();
});
