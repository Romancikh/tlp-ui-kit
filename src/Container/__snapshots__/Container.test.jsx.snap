// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default container renders correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-171i2kh-MuiContainer-root"
  >
    Some content
  </div>
</div>
`;

exports[`MaxWidth container renders correctly 1`] = `
<div>
  <div
    class="MuiContainer-root css-14v630a-MuiContainer-root"
  >
    Some content
  </div>
</div>
`;
